client/my-sites/stats/controller.jsx

```diff
@@ -268,8 +265,9 @@ module.exports = {
 		} else if ( ! activeFilter || -1 === validModules.indexOf( context.params.module ) ) {
 			next();
 		} else {
-			if ( 'object' === typeof( site.options ) && 'undefined' !== typeof( site.options.gmt_offset ) ) {
-				momentSiteZone = i18n.moment().utcOffset( site.options.gmt_offset );
+			const gmtOffset = getSiteOption( context.store.getState(), siteId, 'gmt_offset' );
+			if ( Number.isFinite( gmtOffset ) ) {
```

Comment:
In the ternary case for 0 value (false) we apply 0 anyway, the result of using just truth verification is the same. I thought the extra call would to `Number.isFinite` would be irrelevant.  But thinking better, you are right we should use the same condition, so I updated the ternary to use `Number.isFinite`.